runtime/lua/vim/treesitter/_fold.lua

```diff
@@ -233,6 +233,10 @@ local M = {}
 local foldinfos = {}
 
 local function recompute_folds()
+  if vim.wo.foldmethod ~= 'expr' then
```

Comment:
Can you add a comment saying that foldexpr could have changed since this function is scheduled.

Reply:
Does the callback to the autocmd below also need the same guard?

Reply:
> Does the callback to the autocmd below also need the same guard?

i think so. will add soon.

btw there is another case that could go wrong: when text changes in a buffer that is not shown in the current window.